src/Adapter/CartRule/CartRuleBuilder.php

```diff
@@ -55,6 +56,13 @@ public function build(AddDiscountCommand $command): CartRule
         $cartRule->type = $command->getDiscountType()->getValue();
         $cartRule->free_shipping = $command instanceof AddFreeShippingDiscountCommand;
 
+        if ($command instanceof AddCartLevelDiscountCommand) {
+            $cartRule->reduction_percent = (float) (string) $command->getPercentDiscount();
+            $cartRule->reduction_amount = (float) (string) $command->getAmountDiscount()?->getAmount();
+            $cartRule->reduction_currency = $command->getAmountDiscount()?->getCurrencyId()->getValue();
+            $cartRule->reduction_tax = $command->getAmountDiscount()?->isTaxIncluded();
```

Comment:
Since both values are contradictory when one is set the other one should be reset:
```suggestion
            if ($command->getPercentDiscount()) {
                $cartRule->reduction_percent = (float) (string) $command->getPercentDiscount();
                $cartRule->reduction_amount = 0;
                $cartRule->reduction_currency = 0;
                $cartRule->reduction_tax = false;
            } elseif ($command->getAmountDiscount()) {
                $cartRule->reduction_percent = 0;
                $cartRule->reduction_amount = (float) (string) $command->getAmountDiscount()->getAmount();
                $cartRule->reduction_currency = $command->getAmountDiscount()->getCurrencyId()->getValue();
                $cartRule->reduction_tax = $command->getAmountDiscount()->isTaxIncluded();
            }
```